docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,64 @@
+# Adding a New Metrics Provider for Dummy Endpoint
+
+To create a provider for the dummy endpoint, follow these steps:
+
+1. Fork the [Keptn repo](https://github.com/keptn/lifecycle-toolkit)
+
+2. **Define the Provider Type:** In the `metrics-operator/controllers/common/providers/common.go` file,
+ define the constant `KeptnDummyProviderType` with the value `"dummy"`.
+
+    ```go
+    const KeptnDummyProviderType = "dummy"
+    ```
+
+3. **Implement the Provider:** Create your own new folder inside the
+[metrics-operator/controllers/common/providers](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/controllers/common/providers)
+ matching the new service name: dummy and a new Go package for the dummy provider.
+  This package should contain
+ a struct that implements the `KeptnSLIProvider` interface.
+  To fully implement the `KeptnSLIProvider` interface, it's necessary to implement the
+  `EvaluateQuery`, `EvaluateQueryForStep` and `FetchAnalysisValue` functions.
+  You can follow other existing implementations,
+ such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+ as an example.
+  Below is an example of a dummy provider implementation.
+
+    ```go
+    {% include "./dummy-code-example.go" %}
```

Comment:
togheter with the dummy code with the strings we are working on, here I would also spend a few words explaining what is the use of each function. Especially important is to refer to the documentation of metrics and analysis, to hel the user understand what they should retrieve from the methods inputs to compute their results :)